crates/red_knot_wasm/src/lib.rs

```diff
@@ -386,6 +414,21 @@ impl Range {
         let end = line_index.source_location(text_range.end(), source);
         Self::from((start, end))
     }
+
+    fn to_text_range(self, line_index: &LineIndex, source: &str) -> ruff_text_size::TextRange {
+        let start = line_index.offset(
+            OneIndexed::new(self.start.line).unwrap(),
+            OneIndexed::new(self.start.column).unwrap(),
+            source,
+        );
+        let end = line_index.offset(
+            OneIndexed::new(self.end.line).unwrap(),
+            OneIndexed::new(self.end.column).unwrap(),
+            source,
+        );
```

Comment:
We should make this return a `Result` instead of panicking if the line number is one indexed. 

We can then also use it on line 261, and 219